File: web-services/src/main/java/com/eslam/webservices/service/LocationServiceImpl.java
package com.eslam.webservices.service;

import com.eslam.webservices.entity.Location;
import com.eslam.webservices.reposatory.LocationRepository;
import org.springframework.stereotype.Service;

import java.util.List;
import java.util.Optional;

@Service
public class LocationServiceImpl implements LocationService {

    final
    LocationRepository locationRepository;

    public LocationServiceImpl(LocationRepository locationRepository) {
        this.locationRepository = locationRepository;
    }

    public List<Location> retrieveLocations() {
        return (List<Location>) locationRepository.findAll();
    }

    @Override
    public Location retrieveLocationsById(Long id) {

        Optional<Location> location = locationRepository.findById(id);
        return location.get();
    }
}

File: web-services/src/main/java/com/eslam/webservices/service/LocationServiceImpl.kt
package com.eslam.webservices.service

import com.eslam.webservices.entity.Location
import com.eslam.webservices.reposatory.LocationRepository
import com.eslam.webservices.service.LocationService
import org.springframework.stereotype.Service

@Service
class LocationServiceImpl(private val locationRepository: LocationRepository) : LocationService {
    override fun retrieveLocations(): List<Location> {
        return locationRepository.findAll() as List<Location>
    }

    override fun retrieveLocationsById(id: Long): Location {
        val location = locationRepository.findById(id)
        val save = locationRepository.save(location.get())
        return location.get()
    }
}